draft-gpew-priv-ppm.md

```diff
@@ -585,6 +586,13 @@ type 'outdatedConfig'. Clients SHOULD invalidate any cached aggregator
 `HpkeConfig` and retry with a freshly generated Report. If this retried report
 does not succeed, clients MUST abort and discontinue retrying.
 
+The leader MUST ignore any report whose timestamp falls in a batch interval that
+has been agregated at least once. Otherwise, comparing the aggregate result to
```

Comment:
What does it mean for a *batch interval* to have been *aggregated*? I don't think that verb is defined for that noun in the protocol right now. As a reader, if I want to see where things become "aggregated", then the `AggregateReq/AggregateResp` exchange seems like the obvious place, but an `AggregateReq` is not defined in terms of a batch interval. `AggregateSubReq` does include parameters that uniquely identify a `Report`, which I think lets us conclude that a `Report` is aggregated when the leader gets an `AggregateSubResp` from the helper for a `Report`. `OutputShareReq` does have a `batch_interval` member, so is the receipt of the corresponding `EncryptedOutputShare` when a batch interval becomes "aggregated"?

I think we should write out a state machine for reports and batch intervals, especially since it seems like they enter the "aggregated" state at different moments. We maybe want to use a different verb than "aggregate" for batch interval to avoid confusion, too. How about "collect", since the `OutputShareReq/EncryptedOutputShare` process should be triggered by a collector issuing `CollectReq` to leader?

Also I think we should not do something like add a `batch_interval` member to `AggregateReq` and declare that receipt of `AggregateResp` is when a batch interval is aggregated, because I think that would prevent leaders and helpers from summing Prio inputs ahead of receipt of all the reports in a batch interval.

Reply:
Why are you saying "ignore" as opposed to "generate error"

Reply:
@tgeoghegan good call. I've changed "aggregated" to be more precise. (Similarly below.)

Reply:
@ekr good question.

One reason to not *require* an error here is that, if we did, the leader would have to check if the batch has been collected before completing the HTTP request. The current text allows it to make this determination later on.

One reason to *not permit* an error at all is that it volunteers information to the client that the client doesn't really need, namely, that the collector has requested the aggregate result for the interval. This doesn't seem all that damaging right now, but who knows.

Reply:
And one reason to provide an error is so clients can debug themselves.

Reply:
The bug in this case is clockskew, right?

Reply:
Or any number of other things that cause the clock to be wrong.

Reply:
In any case, the benefit seems marginal to me. But I don't feel that strongly about it. I'd go for permitting an error here if you'd prefer that.

Reply:
I see no reason to just let this behavior go unspecified. What if some leaders reply with 200 OK and some reply with an error in this case? At the end of the day, the report upload was unsuccessful, and the signal sent back to the client should not be 200 OK.

Reply:
@chris-wood supports the idea of having the leader issue an error here. Quote: 
> I agree, and I think we need to be clear with the protocol interaction here. I am not supportive of "ignore behavior is left unspecified."

As I mentioned above, it may not always be feasible for the leader to check for the error condition on-the-fly. It requires less internal-state coordination to just ingest the report and prune it later. I am fine with a MAY or even a SHOULD here but I don't think it should be a MUST. I'm fine being in the rough, just let me know if you think this absolutely needs to be a MUST.

Reply:
Sure, I'm operating under the assumption that the leader can detect this immediately upon upload. If the leader defers the check to some point later on, then obviously it can't reply with an error on the fly. A SHOULD seems like it'd work here.